### PR TITLE
Ensure compliant init options API implementations.

### PR DESCRIPTION
**Pull request by hidmic**
Complying with https://github.com/ros2/rmw/pull/244.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `rcl` and `test_rmw_implementation` against all RMW implementations:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11262/)

File: rmw_connext_cpp/src/rmw_init.cpp [hunk 1]
```diff
@@ -64,39 +68,46 @@ rmw_init_options_copy(const rmw_init_options_t * src, rmw_init_options_t * dst)
...
-  allocator.deallocate(init_options->enclave, allocator.state);
-  rmw_security_options_fini(&init_options->security_options, &allocator);
-  *init_options = rmw_get_zero_initialized_init_options();
-  return RMW_RET_OK;
+  const rcutils_allocator_t * allocator = &init_options->allocator;
+  RCUTILS_CHECK_ALLOCATOR(allocator, return RMW_RET_INVALID_ARGUMENT);
+
+  rmw_ret_t ret = rmw_security_options_fini(&init_options->security_options, allocator);
+  if (RMW_RET_OK == ret) {
+    allocator->deallocate(init_options->enclave, allocator->state);
```
Comment (ivanpauno) [inline rmw_connext_cpp/src/rmw_init.cpp:107]: see https://github.com/ros2/rmw_fastrtps/pull/399#discussion_r444441126

File: rmw_connext_cpp/src/rmw_init.cpp [hunk 2]
```diff
@@ -64,39 +68,46 @@ rmw_init_options_copy(const rmw_init_options_t * src, rmw_init_options_t * dst)
...
+    RMW_SET_ERROR_MSG("expected initialized init_options");
+    return RMW_RET_INVALID_ARGUMENT;
+  }
   RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
     init_options,
     init_options->implementation_identifier,
     rti_connext_identifier,
     return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
-  allocator.deallocate(init_options->enclave, allocator.state);
-  rmw_security_options_fini(&init_options->security_options, &allocator);
```
Comment (ivanpauno) [inline rmw_connext_cpp/src/rmw_init.cpp:97]: unrelated, but we need to add RMW_WARN_UNUSED to this function declaration

**Comment by hidmic** [comment]
One last sanity CI run after API contract updates:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11281/)

**Pull request merged**